src/js/step.js

```diff
@@ -322,32 +310,91 @@ export class Step extends Evented {
     }
   }
 
-  destroy() {
-    if (_.isElement(this.el) && this.el.parentNode) {
-      this.el.parentNode.removeChild(this.el);
-      delete this.el;
+  /**
+   * Sets the options for the step, maps `when` to events, sets up buttons
+   * @param {Object} options The options for the step
+   */
+  setOptions(options = {}) {
+    this.options = options;
+    const { when } = this.options;
+
+    this.destroy();
+    this.id = this.options.id || this.id || `step-${uniqueId()}`;
+
+    _.forOwn(when, (handler, event) => {
+      this.on(event, handler, this);
+    });
+
+    this._setUpButtons();
+  }
+
+  /**
+   * Wraps `_show` and ensures `beforeShowPromise` resolves before calling show
+   * @return {*}
```

Comment:
does esdoc not support `{Promise}` here as the returned value?

Reply:
It does, we don't always return a promise though, if `beforeShowPromise` is not a function.